### PR TITLE
v0.4.0: prepare new release

### PR DESCRIPTION
**Pull request by eulertour**
Prepare for v0.4.0 release.

As an aside, I think we're currently doing too much editing on the changelog. I'd consider the benefit of automating our releases to be greater than that of having an edited changelog.

### CODE REVIEW

File: docs/source/changelog.rst [hunk 1]
```diff
@@ -17,6 +17,63 @@ Upcoming release
...
+- :pr:`1032`: Remove Square.side_length property
+- :pr:`1031`: fix link to wikipedia vector graphics page
+- :pr:`1021`: Docs: added Bezier example
+- :pr:`1017`: Added progress_bar to digest_args to fix the --progress_bar CLI flag
+- :pr:`1018`: Remove redundancy in FunctionGraph arguments
+- :pr:`1024`: Migrate width/height/depth to properties
+- :pr:`1022`: Fix -p flag when passing -s
+- :pr:`1008`: ci: fix release asset upload
+- :pr:`983`: make sure last frame for animations with updaters is correct
+- :pr:`984`: Add manim version to CLI output, append version name for generated gif+png, add version to metadata of rendered videos
```
Comment (WampyCakes) [inline docs/source/changelog.rst:65]:
```suggestion
- :pr:`984`: Add manim version to CLI output, append version name for generated gif+png, add version to metadata of rendered videos, change dark blue terminal text to default green
```

**Comment by WampyCakes** [comment]
#981 will also need added to the changelog once it's merged

**Review by naveen521kk — APPROVED** [review]

File: docs/source/changelog.rst [hunk 2]
```diff
@@ -17,6 +17,63 @@ Upcoming release
...
+- :pr:`1015`: Documentation: Add more explicit instructions related to tlmgr
+- :pr:`1028`: Docs: Update installation guide on mac with Apple Silicon
+- :pr:`1032`: Remove Square.side_length property
+- :pr:`1031`: fix link to wikipedia vector graphics page
+- :pr:`1021`: Docs: added Bezier example
+- :pr:`1017`: Added progress_bar to digest_args to fix the --progress_bar CLI flag
+- :pr:`1018`: Remove redundancy in FunctionGraph arguments
+- :pr:`1024`: Migrate width/height/depth to properties
+- :pr:`1022`: Fix -p flag when passing -s
+- :pr:`1008`: ci: fix release asset upload
```
Comment (naveen521kk) [inline docs/source/changelog.rst:63]: Shouldn't this be under other changes or something?

**Review by jsonvillanueva — APPROVED** [review]

File: docs/source/changelog.rst [hunk 3]
```diff
@@ -17,6 +17,63 @@ Upcoming release
...
+============
+
+- :pr:`1026`: Add 3D Mobjects - Cone, Cylinder, Line3D, Arrow3D and Torus.
+- :pr:`1047`: add api docs and examples for Matrix mobjects
+- :pr:`1044`: register_font is available for macOS
+- :pr:`995`: Add generic set method and compatibility layer between properties and get_*/set_* methods
+
+Bugfixes and Enhancements
+=========================
+
```
Comment (jsonvillanueva) [inline docs/source/changelog.rst:38]:
```suggestion
- :pr:`981`:  Bugfix: Fix WebGL renderer hot reload on Windows
```

**Review by jsonvillanueva — CHANGES_REQUESTED** [review]

**Comment by jsonvillanueva** [comment]
> 
> As an aside, I think we're currently doing too much editing on the changelog. I'd consider the benefit of automating our releases to be greater than that of having an edited changelog.

Does this mean that the squash/merger needs better final commit message, or something more?

**Comment by behackl** [comment]
Thank you for preparing the release!

> As an aside, I think we're currently doing too much editing on the changelog. I'd consider the benefit of automating our releases to be greater than that of having an edited changelog.

I disagree. I've been away for, like, 10 days or so -- and didn't get at all what some of the entries in the changelog actually mean. I consider myself somewhat involved when it comes to the community edition, but if the changelog doesn't let me easily get back to the current status, how is it going to help other users that are not that familiar with Manim?

This is why I like to put in the work of slightly editing the changelog. Stuff like including links helps a lot, and also replacing the often very unhelpful merge messages by the oneline summaries prepared by the PR authors helps to make changes more transparent and easier to understand.

As long as the people merging the PRs don't write the merge commit messages more careful, I don't see a way around someone having to go through the effort of editing the changelog manually.

I've now also done it for this changelog, but for some reason I am, once again, not allowed to push to your branch. Please use the latest commit from <https://github.com/behackl/manim/tree/prepare-v0.4.0>, which is <https://github.com/behackl/manim/commit/838f946fe01be35870087f249497970f7384abb5>.

**Comment by naveen521kk** [comment]
I think `Rework of SVG engine` is missing.
A breaking change is missing https://github.com/ManimCommunity/manim/wiki/Changelog-for-next-release#breaking-changes

I remember I updated that wiki as and when I merged any PR, so having a look over there should be great.

**Comment by behackl** [comment]
> I think `Rework of SVG engine` is missing.

915 is included in the list (although quite hidden...)

> A breaking change is missing https://github.com/ManimCommunity/manim/wiki/Changelog-for-next-release#breaking-changes
> 

In which PR was this introduced?

**Comment by naveen521kk** [comment]
> In which PR was this introduced

In https://github.com/ManimCommunity/manim/pull/915

> 915 is included in the list (although quite hidden...)

I would say that should be the highlight of this version and should be visible...

**Comment by behackl** [comment]
I agree. I've updated my branch: https://github.com/behackl/manim/commit/8225e1babe961104a6305a54d453107141bb03be

**Comment by WampyCakes** [comment]
behackl@1ca0537 now @eulertour

**Comment by eulertour** [comment]
Closing in favor of #1072

**Pull request closed without merging**